documentation/models.py

```diff
@@ -299,11 +307,19 @@ def _append_suffix_to_app(self, suffix):
                 app_with_suffix = 'https://%s/%s' % (re.search(re_url, self.app).group(0), suffix)
                 return app_with_suffix
 
+    # this legacy code is kept to support data in the database that might expect it.
+    # ideally would use something closer to the code in 'get_embed_app_and_code'
+    # rather than the complicated regex munging below
     def get_embed_app(self):
-        return self._append_suffix_to_app('embed')
+         if self.app:
+            re_url = '\w*(studio.code.org\/p\w*\/\w+\/\w+)'
+            if re.search(re_url, self.app):
+                app_with_suffix = 'https://%s/%s' % (re.search(re_url, self.app).group(0), 'embed')
+                return app_with_suffix
```

Comment:
can this be deduplicated?
```suggestion
         return self._append_suffix_to_app('embed')
```

Reply:
Not entirley sure what you mean by deduplicated here. I am going to hazard a guess that you're asking of we can combine the code for get_embed_app and get_embed_app_and_code.

I don't think this would be a good idea, because get_embed_app_and_code needs a different implementation to support special characters, but get_embed_app is supporting existing data in the database that I am concerned might cause bugs if we were to change it

Reply:
I'm talking about calling `_append_suffix_to_app`. I made a github suggestion in my original comment to show it specifically.